packages/utils/test/route.test.js

```diff
@@ -237,4 +237,109 @@ describe('util: route', () => {
       expect(routesResult).toMatchSnapshot()
     })
   })
+
+  describe('util: route sortRoutes', () => {
+    const files = [
+      'pages/_param.vue',
+      'pages/de/index.vue',
+      'pages/index.vue',
+      'pages/*.vue',
+      'pages/another_route/rout*.vue',
+      'pages/snake_case_route.vue',
+      'pages/subpage/_param.vue',
+      'pages/de*.vue',
+      'pages/parent/index.vue',
+      'pages/parent/*.vue',
+      'pages/another_route/_id.vue',
+      'pages/another_route/rout*.vue',
+      'pages/parent/child/index.vue',
+      'pages/parent/child/test.vue'
+    ]
+    const srcDir = '/some/nuxt/app'
+    const pagesDir = 'pages'
+
+    test.posix('sortRoutes should sort routes', () => {
+      const routesResult = createRoutes({ files, srcDir, pagesDir })
+      expect(routesResult).toMatchSnapshot()
+    })
+    test('Should sortRoutes with extendRoutes using *', () => {
+      const routes = [
+        { path: '/poetry' },
+        { path: '/reports' },
+        { path: '*' },
+        { path: '/de/about' },
+        { path: '/' },
+        { path: '/about' },
+        { path: '/de' },
+        { path: '/tech' },
+        { path: '/de/tech' },
+        { path: '/de*' },
+        { path: '/:post' },
+        { path: '/de/:post' },
+        { path: '/de/reports' },
+        { path: '/de/poetry' }
+      ]
+
+      sortRoutes(routes)
+
+      expect(routes).toEqual(
+        [
+          { path: '/about' },
+          { path: '/de' },
+          { path: '/poetry' },
+          { path: '/reports' },
+          { path: '/tech' },
+          { path: '/de/about' },
+          { path: '/de/poetry' },
+          { path: '/de/reports' },
+          { path: '/de/tech' },
+          { path: '/' },
+          { path: '/de/:post' },
+          { path: '/:post' },
+          { path: '/de*' },
```

Comment:
Looks like `/de*` would be inaccessible this way? Should be before `/:post`?

Reply:
Nice catch, yeah, the /de* won't work on this scenario. Not sure how relevant is this to the use case, but I'll try to fix it and have the same output as the `Should sortRoutes with extendRoutes using /*` test case.